### PR TITLE
HDDS-8722. Improve thread names in Datanode.

### PR DESCRIPTION
**Pull request by Galsza**
## What changes were proposed in this pull request?

MiniOzoneCluster test logs are hard to follow because earlier they didn't contain the thread names. This change fixes it for the Datanode side by adding a prefix that identifies it.

This is a smaller task under a series of tasks [improve thread names](https://issues.apache.org/jira/browse/HDDS-8719)
 
## How was this patch tested?
Run existing unit/integration test through github: https://github.com/Galsza/ozone/actions/runs/6639296207
Run docker based tests and observed logs.

### CODE REVIEW

**Review by adoroszlai — COMMENTED** [review]
Thanks @Galsza for the patch, mostly looks good.

`EndpointStateMachine task thread` has no prefix.  Please check if you can pass the prefix via `StateContext`.

Also this one:

https://github.com/apache/ozone/blob/4fe7824f9fe3923e4a82cd69321b2ffd2196b9ed/hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/OzoneContainer.java#L297-L298

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/OzoneContainer.java [hunk 1]
```diff
@@ -296,8 +296,10 @@ private void buildContainerSet() {
     ContainerInspectorUtil.load();
     while (volumeSetIterator.hasNext()) {
       StorageVolume volume = volumeSetIterator.next();
-      Thread thread = new Thread(new ContainerReader(volumeSet,
-          (HddsVolume) volume, containerSet, config, true));
+      ContainerReader containerReader = new ContainerReader(volumeSet,
+          (HddsVolume) volume, containerSet, config, true);
+      Thread thread = new Thread(containerReader,
+          datanodeDetails.threadNamePrefix());
```
Comment (adoroszlai) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/OzoneContainer.java:302]: Each thread should have some unique ID in addition to the prefix.  I think we can achieve that most easily by creating a thread factory outside of the loop:

```
    String threadNamePrefix = datanodeDetails.threadNamePrefix();
    ThreadFactory threadFactory = new ThreadFactoryBuilder()
        .setDaemon(true)
        .setNameFormat(threadNamePrefix + "ContainerReader-%d")
        .build();
```

then use it to create the threads in the loop:

```
      Thread thread = threadFactory.newThread(containerReader);
```

**Review by adoroszlai — COMMENTED** [review]
Thanks @Galsza for updating the patch.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/OzoneContainer.java (hunk 1 above)
Comment (Galsza) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/OzoneContainer.java]: Thank you @adoroszlai I've changed it. Please take another look

**Review by adoroszlai — APPROVED** [review]

**Comment by Galsza** [comment]
@adoroszlai TestRootCaRotationPoller failed. I've run that test locally 2000 times and didn't see a failure for the same reason as the failing unit test here.

However the tests became slow and after trying again with 10.000 runs the following error happened: `Exception in thread "RootCaRotationPoller" java.lang.OutOfMemoryError: GC overhead limit exceeded` I assume this means the test has some form of memory leak or IntelliJ couldn't handle 10000 test runs cached. But all in all it seems unrelated to the threadnames change. I have never seen this test fail another time, but I see why it could be flaky. Shall we create an issue for it?

**Comment by adoroszlai** [comment]
> I have never seen this test fail another time

Failed here:

* https://github.com/adoroszlai/ozone-build-results/blob/master/2023/10/13/25988/unit/hadoop-hdds/framework/org.apache.hadoop.hdds.security.x509.certificate.utils.TestRootCaRotationPoller.txt
* https://github.com/adoroszlai/ozone-build-results/blob/master/2023/10/25/26238/unit/hadoop-hdds/framework/org.apache.hadoop.hdds.security.x509.certificate.utils.TestRootCaRotationPoller.txt

> Shall we create an issue for it?

Yes, please.

**Comment by Galsza** [comment]
Created https://issues.apache.org/jira/browse/HDDS-9548.

**Comment by adoroszlai** [comment]
Thanks @Galsza, @ArafatKhan2198 for the patch.
